lib/src/components/NxDrawer/NxDrawer.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useEffect, useRef } from 'react';
+import classnames from 'classnames';
+
+import AbstractDialog from '../AbstractDialog/AbstractDialog';
+import NxCloseButton from '../NxCloseButton/NxCloseButton';
+import withClass from '../../util/withClass';
+
+import { Props, propTypes } from './types';
+
+import './NxDrawer.scss';
+
+const DRAWER_OPEN_CLASS_NAME = 'nx-drawer--open';
+
+const _NxDrawer = (props: Props) => {
+  const {
+    className,
+    onCancel,
+    children,
+    variant,
+    headerTitle,
+    headerSubtitle,
+    headerParagraph,
+    ...attrs
+  } = props;
+
+  const dialogRef = useRef<HTMLDialogElement>(null);
+
+  useEffect(() => dialogRef.current?.classList.add(DRAWER_OPEN_CLASS_NAME), [dialogRef]);
```

Comment:
`dialogRef` never changes.  You might have meant` dialogRef.current`, but that also won't change for slightly different reasons.

```suggestion
  useEffect(() => dialogRef.current?.classList.add(DRAWER_OPEN_CLASS_NAME), []);
```